### DISCUSSION TITLE
My Discord Buttons function cannot identify the command i want it to identify

### DISCUSSION THREAD

**Discussion by ghost** [ASEAN-Build-The-Earth/Merlion#64] [Q&A] [answered]
I followed the discord.js v13 guide on discord buttons and i dont think it can identify the file in order to execute the button function.

Index.js code sample 
```
client.on('interactionCreate', async interaction => {
	if (!interaction.isCommand()) return;

	if (interaction.commandName === 'ping') {
		const row = new MessageActionRow()
			.addComponents(
				new MessageButton()
					.setCustomId('primary')
					.setLabel('Primary')
					.setStyle('PRIMARY'),
			);

		await interaction.reply({ content: 'Pong!', components: [row] });
	}
});
```

**Comment by StoneMcYT** [comment]
> I followed the discord.js v13 guide on discord buttons and i dont think it can identify the file in order to execute the button function.

Send stack trace and your code?

**Reply by ghost** [reply]
ok

**Reply by tintinkung** [reply]
yus share ur code maybe i can tri help

**Reply by ghost** [reply]
okay
```js
client.on('interactionCreate', async interaction => {
	if (!interaction.isCommand()) return;

	if (interaction.commandName === 'ping') {
		const row = new MessageActionRow()
			.addComponents(
				new MessageButton()
					.setCustomId('primary')
					.setLabel('Primary')
					.setStyle('PRIMARY'),
			);

		await interaction.reply({ content: 'Pong!', components: [row] });
	}
});
```

**Reply by tintinkung** [reply]
where in file u put that function btw

**Reply by ghost** [reply]
um index.js

**Reply by tintinkung** [reply]
uh

**Comment by StoneMcYT** [comment]
> I followed the discord.js v13 guide on discord buttons and i dont think it can identify the file in order to execute the button function.
> 
> Index.js code sample
> 
> ```
> client.on('interactionCreate', async interaction => {
> 	if (!interaction.isCommand()) return;
> 
> 	if (interaction.commandName === 'ping') {
> 		const row = new MessageActionRow()
> 			.addComponents(
> 				new MessageButton()
> 					.setCustomId('primary')
> 					.setLabel('Primary')
> 					.setStyle('PRIMARY'),
> 			);
> 
> 		await interaction.reply({ content: 'Pong!', components: [row] });
> 	}
> });
> ```

This interesting may I know why there are buttons in a ping command?

**Reply by tintinkung** [reply]
dat prop just testing bruh

**Reply by ghost** [reply]
i am doing a test first before doing the actual code i want to test it so that i get the general idea

**Comment by tintinkung** [comment, marked as answer]
@S4ge-Rites try create this in new file
u cant just copy and paste example file from discord.js guide our code doesnt work like the guide one, we read message and args seperately
try read this code and understand:
```js
const { MessageButton, MessageActionRow } = require("discord.js");

module.exports = {
    name: 'test1',
    description: 'random message',
    execute(message, args) 
    {
        const row = new MessageActionRow()
			.addComponents(
            new MessageButton()
                .setLabel("uwu")
                .setCustomId('primary')
                .setLabel('Primary')
                .setStyle('PRIMARY')
                .setDisabled(true)
            );

        message.reply({ content: 'Pong!', components: [row] });
    },
};
```

**Reply by ghost** [reply]
Thank you so much the code worked
